src/video/SDL_video.c

```diff
@@ -1949,6 +1949,8 @@ bool SDL_UpdateFullscreenMode(SDL_Window *window, SDL_FullscreenOp fullscreen, b
             SDL_MinimizeWindow(display->fullscreen_window);
         }
 
+        display->fullscreen_active = true;
```

Comment:
Basically the only change required is:
```suggestion
        display->fullscreen_active = window->fullscreen_exclusive;
```
so that if the user externally changes the desktop resolution while a fullscreen desktop window is active, the desktop resolution is still updated, while changes from mode switches that come from SDL for exclusive fullscreen windows are dropped.

Reply:
Oh, that's great. Added!